### ISSUE TITLE
Pretrained models do not work with sampling rates different from the one used for training

### ISSUE THREAD

**Issue by terrykwon** [pyannote/pyannote-audio#289]
I wanted to run the pretrained SAD on a single test file. I got back valid results, but the returned `extent` duration seems to be less than 1/3rd of the original file: `sad_scores.extent` is 14 minutes, while the original audio file is around 40 minutes.

```python
sad = torch.hub.load('pyannote/pyannote-audio:develop', 'sad_ami')
test_file = {'uri': 'filename',
             'audio': '/data/speech/test.wav'}
sad_scores = sad(test_file) # array of [float, float]
print(sad_scores.extent) # [ 00:00:00.000 -->  00:14:28.619]
```

Am I loading the file wrong? I looked in the documentation but could not find a guide to feeding the model a single .wav file, except for the code above.

**Comment by hbredin** [comment]
It does not look like you are doing something wrong: this is the way to go.

Can you please share the output of this script? 

```python
>>> test_file = {'audio': '/data/speech/test.wav'}

>>> from pyannote.audio.features.utils import read_audio
>>> y, sample_rate = read_audio(test_file)
>>> print(f'{len(y)} samples at {sample_rate}Hz')

>>> from pyannote.audio.features.utils import get_audio_duration
>>> duration = get_audio_duration(test_file)
>>> print(f'{duration:.3f} seconds')
```

Or (even better) share the `test.wav` itself?

**Comment by terrykwon** [comment]
@hbredin thank you for the quick response. I realized the problem: the file had a sampling rate of 44.1kHz. Downsampling to 16kHz solved the issue.

Maybe you could add somewhere in the documentation that the input to the pretrained models should be 16kHz?

```python
from pyannote.audio.features.utils import read_audio
y, sample_rate = read_audio(test_file)
print(f'{len(y)} samples at {sample_rate}Hz') # 105580162 samples at 44100Hz

from pyannote.audio.features.utils import get_audio_duration
duration = get_audio_duration(test_file)
print(f'{duration:.3f} seconds') # 2394.108 seconds
```

**Comment by hbredin** [comment]
Actually, this is a bug: it should work for any sampling rate. I'll look into it.
In the meantime, downsampling to 16kHz should indeed circumvent the problem.

**Comment by hbredin** [comment]
I think I fixed the bug. 

Can you please try again with the latest commit in `develop` branch.

If you provide a file in a different sample rate (like the one you used with 44.1kHz), it should resample it on-the-fly and return the correct extent.

I'd appreciate if you could confirm the bug is now fixed.

**Comment by hbredin** [comment]
By the way, I see that you are using `sad_ami` model. 

I don't what kind of data you have but using `sad_dihardx` might give you better results.
It is the model described in [this paper](https://arxiv.org/abs/1910.10655).

**Comment by terrykwon** [comment]
@hbredin Yep, I can confirm the issue is fixed. I am getting identical results for my 16kHz and 44.1kHz file.

Thanks for all the work you're putting into this awesome library! Plus thanks for the tip—it made a significant difference (10+% accuracy improvement)👍

**Comment by hbredin** [comment]
Great. Thanks for the feedback (and for finding a bug!)

If that is not too much to ask, could you please share
- the numbers (accuracy, detection error rate) you got with sad_ami vs. sad_dihardx
- the type of audio recording you are using (phone calls? meetings? indoor? outdoor? background noise?)

**Comment by terrykwon** [comment]
@hbredin I haven't fine-tuned the binarizer yet (`binarize = Binarize(offset=0.52, onset=0.52, log_scale=True, min_duration_off=0.1, min_duration_on=0.1)`), but the current results are:

* detection accuracy: 0.72 (`sad_ami`) vs. 0.85 (`sad_dihard`)
* detection error rate: 0.46 vs. 0.24

Note that this is for a single 40-min file only. It is a conversation between an infant and a mother (indoors, noisy), so it was expected that a pretrained model would give low accuracy. I'm afraid I cannot share the audio at this point since it was recorded for an experiment.

I've also attached the raw SAD scores in a 1-minute timeframe for reference.

![download](https://user-images.githubusercontent.com/7956158/73538534-a4313100-446e-11ea-841d-f9e8842e43ff.png)

**Comment by hbredin** [comment]
Thanks for this very detailed answer.

May I suggest you try `sad_dihardx` (with an `x` at the end) instead of `sad_dihard`?
`sad_dihardx` should give you even better results as it was trained to be less sensitive to domain mismatch.

**Comment by terrykwon** [comment]
@hbredin Sorry, my mistake. Those results were for `sad_dihardx`.
